cms/djangoapps/import_from_modulestore/models.py

```diff
@@ -0,0 +1,172 @@
+    UsageKeyField,
+)
+from openedx_learning.api.authoring_models import LearningPackage, PublishableEntity
+
+from .data import ImportStatus
+
+User = get_user_model()
+
+
+class Import(TimeStampedModel):
+    """
+    Represents the action of a user importing a modulestore-based course or legacy
+    library into a learning-core based learning package (today, that is always a content library).
+    """
+
+    uuid = models.UUIDField(default=uuid_tools.uuid4, editable=False, unique=True, db_index=True)
+    status = models.CharField(max_length=100, choices=ImportStatus.choices, default=ImportStatus.PENDING)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    source_key = LearningContextKeyField(help_text=_('The modulestore course'), max_length=255, db_index=True)
+    target = models.ForeignKey(LearningPackage, models.SET_NULL, null=True)
+
+    def __str__(self):
+        return f'{self.source_key} - {self.target}'
+
+    def ready(self) -> None:
+        """
+        Set import status to ready.
+        """
+        self.status = ImportStatus.READY
+        self.save()
+
+    def imported(self) -> None:
+        """
+        Set import status to imported and clean related staged content.
+        """
+        self.status = ImportStatus.IMPORTED
+        self.save()
+        self.clean_related_staged_content()
+
+    def cancel(self) -> None:
+        """
+        Cancel import action and delete related staged content.
+        """
+        self.status = ImportStatus.CANCELED
+        self.save()
+        self.clean_related_staged_content()
+
+    def error(self) -> None:
+        """
+        Set import status to error and clean related staged
+        """
+        self.status = ImportStatus.ERROR
+        self.save()
+        self.clean_related_staged_content()
+
+    def clean_related_staged_content(self) -> None:
+        """
+        Clean related staged content.
+        """
+        for staged_content_for_import in self.staged_content_for_import.all():
+            staged_content_for_import.staged_content.delete()
+
+    class Meta:
+        verbose_name = _('Import from modulestore')
+        verbose_name_plural = _('Imports from modulestore')
+
+    @classmethod
+    def get_by_uuid(cls, import_uuid: str) -> Self | None:
+        """
+        Get an import task by its ID.
+        """
+        return cls.objects.filter(uuid=import_uuid).first()
+
+    @classmethod
+    def get_ready_by_uuid(cls, import_uuid: str) -> Self | None:
+        """
+        Get an import task by its UUID.
+        """
+        return cls.objects.filter(uuid=import_uuid, status=ImportStatus.READY).first()
+
+    def get_staged_content_by_block_usage_id(self, block_usage_id: str) -> Optional["StagedContent"]:
+        """
+        Get staged content by block usage ID.
+        """
+        staged_content_for_import = self.staged_content_for_import.filter(
+            staged_content__tags__icontains=block_usage_id
+        ).first()
```

Comment:
This query will be inefficient, due the repeated substring checks. It will also be highly error-prone: imagine a scenario where the block usage id is `chemistry-problem`, and there happens to be an unrelated tag on the staged content for `chemistry-problem`.

It would be better to filter using a full usage key string. Obviously that is not available to us on StagedContent model (which is why you are using `icontains` here as a workaround, I imagine). But, we have the new StagedContentForImport model available to us. Would it work if you added a UsageKeyField to that model, and then updated this method to query using that new field?

Reply:
I agree with you that the query is not the most efficient, but we are filtering by the full usage key string (I updated the variable name and docstring so that it doesn't confuse), so the probability of error is minimal, especially since we are filtering only among related `staged_content_for_import` records.

Adding a UsageKeyField to StagedContentForImport won't solve the performance issue, because StagedContentForImport is associated with a single StagedContent record at the chapter level, so if we write its UsageKey, it won't help when filter for blocks at the sequential level and below.

Reply:
Oh, I see, so the staged content contains tags of every single usage key, from the chapter level down to the component level?

Reply:
Yes, the course is stores in StagedContent by sections. So, I don't think filtering by StagedContent in this way will be a problem, since there will be as many import event related records (among which the filtering takes place) as there are sections in the course.
<img width="1524" alt="Знімок екрана 2025-04-16 о 22 01 00" src="https://github.com/user-attachments/assets/07a1ed12-7b4b-4bd4-bc20-1eba9564a44a" />

Reply:
> it won't help when filter for blocks at the sequential level and below.

Why do we need to filter for blocks at the sequential level and below? Would it not be OK if this method only worked with chapter usage keys?

Reply:
Since the import feature involves importing different levels, if the Block usage id of a unit is specified, for example, we need to know which StagedContent record the content of this unit was written to.

Reply:
Got it, thank you for explaining.

I don't believe there is an immediate need to be able to import specific blocks in the hierarchy. We can investigate this possibility with the Product team more in the future, but for Teak, the only product requirement is the ability to import the entire course.

Since this method is inefficient and not strictly necessary, I would prefer if left it out of the model for now. If we do end up needing it, we can work to optimize it. If it is impossible to optimize, then I'd rather it be coded as private helper function--that way, we would not be encouraging other edx-platform developers to make use of it, as we should only be encouraging use of Python APIs which are stable and efficient.

Reply:
Understood, thank you! 
In that case, I've made this method private for now, and added a new field to StagedContentForImport.
I'll add an optimized public method for filtering in the next PR along with adding the Python API.

Reply:
Also, we need to merge this PR as soon as possible to unblock the opening of the next ones. 
I'll open them now, to this taget branch for now, and update when this PR is merged.

Reply:
Please remove it. There is no reason to merge unused code. We can add back some edited version of it if we need it.

Reply:
Removed